tests/test_drivers.py

```diff
@@ -49,5 +52,41 @@ def testMissingSubmitPath(self):
             _init_submission_driver({"payload": {"outputRun": "bad"}})
 
 
+class TestPingDriver(unittest.TestCase):
+    def testWmsServiceSuccess(self):
+        retval = ping_driver("wms_test_utils.WmsServiceSuccess")
+        self.assertEqual(retval, 0)
+
+    def testWmsServiceFailure(self):
+        with self.assertLogs(level=logging.INFO) as cm:
+            retval = ping_driver("wms_test_utils.WmsServiceFailure")
+            self.assertRegex(cm.output[0], "WARN.+service failure")
+            self.assertRegex(cm.output[1], "ERROR.+Couldn't contact service X")
+            self.assertNotEqual(retval, 0)
+
+    def testWmsServiceEnvVar(self):
+        with unittest.mock.patch.dict(
+            os.environ, {"BPS_WMS_SERVICE_CLASS": "wms_test_utils.WmsServiceSuccess"}
+        ):
+            retval = ping_driver()
+            self.assertEqual(retval, 0)
+
+    @unittest.mock.patch.dict(os.environ, {})
+    def testWmsServiceNone(self):
+        # Override default wms to be the test one
+        with unittest.mock.patch.object(BpsConfig, "__getitem__") as mock_function:
+            mock_function.return_value = "wms_test_utils.WmsServiceDefault"
+            with self.assertLogs(level=logging.INFO) as cm:
+                retval = ping_driver()
+                self.assertEqual(retval, 0)
+                self.assertRegex(cm.output[0], "INFO.+DEFAULT")
```

Comment:
If you know you are only getting INFO messages (and you have asserted at least that above) then you can simplify this a bit with:

```
self.assertIn("DEFAULT", cm.output[0])
```
but a regex is also fine.